homeassistant/components/sensor/esphome.py

```diff
@@ -0,0 +1,62 @@
+"""Support for esphome sensors."""
+import logging
+import math
+from typing import Optional, TYPE_CHECKING
+
+from homeassistant.components.esphome import EsphomeEntity, \
+    platform_async_setup_entry
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from aioesphomeapi import SensorInfo, SensorState  # noqa
+
+DEPENDENCIES = ['esphome']
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType,
+                            entry: ConfigEntry, async_add_entities) -> None:
+    """Set up esphome sensors based on a config entry."""
+    # pylint: disable=redefined-outer-name
+    from aioesphomeapi import SensorInfo, SensorState  # noqa
+
+    await platform_async_setup_entry(
+        hass, entry, async_add_entities,
+        component_key='sensor',
+        info_type=SensorInfo, entity_type=EsphomeSensor,
+        state_type=SensorState
+    )
+
+
+class EsphomeSensor(EsphomeEntity):
+    """A sensor implementation for esphome."""
+
+    @property
+    def _static_info(self) -> 'SensorInfo':
+        return super()._static_info
```

Comment:
This doesn't look necessary.

Reply:
For the runtime it's not necessary, yes. 

The reason it's here is because the parent `_static_info` property is type annotated as returning `EntityInfo`. So then my IDE for example highlights the line `self._static_info.unit_of_measurement` as an error, as `EntityInfo` doesn't have `unit_of_measurement`.

As you might have noticed, I like type annotations 😇